src/tests/stress/pedsim/Ped_util/Global_funs.enc

```diff
@@ -0,0 +1,230 @@
+      ""
+    end
+  end
+end
+fun sqrt(nr : real) : real
+  EMBED (unit)
+    { #{nr}=sqrt(#{nr}); }
+  END
+  nr
+end
+fun abs(a : int) : int
+  if a > 0 then
+    a
+  else
+    a * -(1)
+  end
+end
+
+fun round(x : real) : int
+  val i = 0
+  EMBED (unit)
+    { #{i}=round(#{x}); }
+  END
+  i
+end
+
+fun distance_int(x1 : int, y1 : int, x2 : int, y2 : int) : real
+  sqrt(x1 - x2 * x1 - x2 + y1 - y2 * y1 - y2)
+end
+
+fun parse_file(fname : String) : [Agent]
+  var sen = new XML_lib()
+  var file = new XML_node()
+  do
+    match sen.file_to_xml(fname) with
+      case Just(xml) =>
+        file = xml
+      end
+      case Nothing =>
+        print("\n\nBADXML!\n")
+      end
+
+    end
+    var xml_way = file.children_named("waypoint")
+    var waypoints = new [(int, int)](|xml_way|)
+    var waypoints_distance = new [int](|xml_way|)
+    var i = 0
+    do
+      for a <- xml_way do
+        waypoints(i) = match (a.attribute_value("x").to_int(), a.attribute_value("y").to_int()) with
+                         case (Just(x), Just(y)) =>
+                           (x, y)
+                         end
+                         case _ =>
+                           print("BADINT!\n")
+                           abort("parsing error")
+                           (0,0)
+                         end
+
+                       end
+        waypoints_distance(i) = yolo_int(a.attribute_value("r").to_int())
+        i = i + 1
+      end
+      var nr_agents = 0
+      for a <- file.children_named("agent") do
+        nr_agents = nr_agents + match a.attribute_value("n").to_int() with
+                                  case Just(nr) =>
+                                    nr
+                                  end
+                                  case _ =>
+                                    print("BADINT2!\n")
+                                    abort("parsing error")
+                                    0
+                                  end
+
+                                end
+      end
+      var agents = new [Agent](nr_agents)
+      var i = 0
+      do
+        for a <- file.children_named("agent") do
+          var max = yolo_int(a.attribute_value("n").to_int())
+          var x = yolo_int(a.attribute_value("xs").to_int())
+          var y = yolo_int(a.attribute_value("ys").to_int())
+          var dx_max = yolo_int(a.attribute_value("dx").to_int()) / 2
+          var dy_max = yolo_int(a.attribute_value("dy").to_int()) / 2
+          var agent_waypoints = new [(int, int)](|a.children_named("addway")|)
+          var dx = x - dx_max
+          var dy = y - dy_max
+          var j = 0
+          do
+            for b <- a.children_named("addway") do
+              agent_waypoints(j) = waypoints(yolo_int(yolo_string(b.attribute_value("id").substring(1, 2)).to_int()) - 1)
+              unless |agent_waypoints| > 1 then
+                print("THIS SHOULD NEVER HAPPEN |new_agent.waypoint|={}\n", |agent_waypoints|)
```

Comment:
abort does not support parameters, so the int needs to be printer separately anyway.